eyelid-match-ops/src/primitives/yashe/conf.rs

```diff
@@ -25,27 +29,67 @@ use crate::primitives::poly::modular_poly::conf::TinyTest;
 /// Encryption keys and ciphertexts with different parameters are incompatible.
 pub trait YasheConf: PolyConf
 where
-    Self::Coeff: From<u64> + From<i64>,
+    // The `Field` trait is already `From<u128> + From<u64>` (and all the other unsigned types).
+    // The `Fp` types are `From<i64>` (and all the other signed types).
+    // But there are no trait bounds guaranteeing these conversions, so we need to require them.
+    // Unfortunately, these bounds also need to be copied to each generic type and impl block.
+    Self::Coeff: From<u128> + From<u64> + From<i64>,
 {
     /// The plaintext coefficient modulus
     const T: u64;
 
-    /// The standard deviation
-    const DELTA: f64;
+    /// The standard deviation for key generation sampling.
+    /// The default parameters are as recommended in the paper.
+    const KEY_DELTA: f64 = 3.2;
+
+    /// The standard deviation for encryption error sampling
+    /// The default parameters are as recommended in the paper.
+    const ERROR_DELTA: f64 = 1.0;
 
-    /// A convenience method to convert `T` to the `Coeff` type.
+    /// A convenience method to convert [`T`](Self::T) to the [`Coeff`](PolyConf::Coeff) type.
     fn t_as_coeff() -> Self::Coeff {
         Self::Coeff::from(Self::T)
     }
+
+    /// A convenience method to convert [`T`](Self::T) to `u128`.
+    fn t_as_u128() -> u128 {
+        u128::from(Self::T)
+    }
+
+    /// A convenience method to convert a [`Coeff`](PolyConf::Coeff) to `u128`.
+    /// TODO: move this method to a trait implemented on `Coeff` instead.
+    fn coeff_as_u128(coeff: Self::Coeff) -> u128 {
+        let coeff: BigUint = coeff.into();
+
+        coeff
+            .to_u128()
+            .expect("coefficients are small enough for u128")
+    }
+
+    /// A convenience method to convert [`Coeff::MODULUS`](PrimeField::MODULUS) to `u128`.
+    fn modulus_as_u128() -> u128 {
+        let modulus: BigUint = Self::Coeff::MODULUS.into();
+
+        modulus
+            .to_u128()
+            .expect("constant modulus is small enough for u128")
+    }
+
+    /// A convenience method to convert [`Coeff::MODULUS_MINUS_ONE_DIV_TWO`](PrimeField::MODULUS_MINUS_ONE_DIV_TWO) to `u128`.
+    fn modulus_minus_one_div_two_as_u128() -> u128 {
+        let modulus: BigUint = Self::Coeff::MODULUS_MINUS_ONE_DIV_TWO.into();
+
+        modulus
+            .to_u128()
+            .expect("constant modulus is small enough for u128")
+    }
```

Comment:
This conversion was really tricky to find, but it is better to use existing tested methods.